lisa/conf.py

```diff
@@ -420,10 +432,12 @@ class MultiSrcConfMeta(abc.ABCMeta):
     def __new__(metacls, name, bases, dct, **kwargs):
         new_cls = super().__new__(metacls, name, bases, dct, **kwargs)
         if not inspect.isabstract(new_cls):
-            doc = new_cls.__doc__
-            if doc:
-                # Create a ResStructuredText preformatted block
-                generated_help = '\n' + new_cls.get_help(style='rst')
+            if new_cls.__doc__:
+                doc = textwrap.dedent(new_cls.__doc__)
+                # Create a ResStructuredText preformatted block when rendering
+                # with Sphinx
+                style = 'rst' if is_running_sphinx() else None
```

Comment:
Seeing as all of our docstrings are sphinx-compliant anyway (or at least strive to be), we could do this unconditionally.

Reply:
The current behavior is to always use reStructuredText. That gives a relatively unusable (lots of empty lines getting in the way) `help()` output, and same goes for shift+<TAB> on a `PlatformInfo` in a notebook. Since we are only really interested in rst in Sphinx context, it was easier to match on Sphinx than on notebook/interactive shell/ipython shell/script/whatever:

```
class PlatformInfo(lisa.conf.MultiSrcConf, lisa.utils.HideExekallID)
 |  PlatformInfo(conf=None, src='user')
 |  
 |      Platform-specific information made available to tests.
 |  
 |      
 |  * platform-info: Platform-specific information
 |  
 |  ..
 |  
 |      * rtapp: RTapp configuration
 |      
 |      ..
 |      
 |          * calib (:class:`~lisa.conf.IntIntDict`): RTapp calibration dictionary
 |      
 |      ..
 |      
 |      * kernel: Kernel-related information
 |      
 |      ..
 |      
 |          * version (:class:`~devlib.target.KernelVersion`)
 |          * config (:class:`~devlib.target.TypedKernelConfig`)
 |      
 |      ..
 |      
 |      * nrg-model (:class:`~lisa.energy_model.EnergyModel`): Energy model object
 |      * cpu-capacities (:class:`~lisa.conf.IntIntDict`): Dictionary of CPU ID to capacity value
 |      * abi (:class:`~str`): ABI, e.g. "arm64"
 |      * os (:class:`~str`): OS being used, e.g. "linux"
 |      * name (:class:`~str`): Free-form name of the board
 |      * cpus-count (:class:`~int`): Number of CPUs
 |      * freq-domains (:class:`~lisa.conf.IntListList`): Frequency domains modeled by a list of CPU IDs for each domain
 |      * freqs (:class:`~lisa.conf.IntIntListDict`): Dictionnary of CPU ID to list of frequencies
 |      * capacity-classes (:class:`~lisa.conf.IntListList`): (derived from platform-info/cpu-capacities) Capacity classes modeled by a list of CPU IDs for each capacity, sorted by capacity
 |  
 |  ..
```

Reply:
With that PR, it gives a more compact representation, that fits in the shift+TAB tooltip of jupyter:
```
class PlatformInfo(lisa.conf.MultiSrcConf, lisa.utils.HideExekallID)
 |  PlatformInfo(conf=None, src='user')
 |  
 |  Platform-specific information made available to tests.
 |  
 |  
 |  +- platform-info: Platform-specific information
 |      +- rtapp: RTapp configuration
 |          |- calib (lisa.conf.IntIntDict): RTapp calibration dictionary.
 |      +- kernel: Kernel-related information
 |          |- version (devlib.target.KernelVersion).
 |          |- config (devlib.target.TypedKernelConfig).
 |      |- nrg-model (lisa.energy_model.EnergyModel): Energy model object.
 |      |- cpu-capacities (lisa.conf.IntIntDict): Dictionary of CPU ID to capacity value.
 |      |- abi (str): ABI, e.g. "arm64".
 |      |- os (str): OS being used, e.g. "linux".
 |      |- name (str): Free-form name of the board.
 |      |- cpus-count (int): Number of CPUs.
 |      |- freq-domains (lisa.conf.IntListList): 
 |         Frequency domains modeled by a list of CPU IDs for each
 |         domain.
 |      |- freqs (lisa.conf.IntIntListDict): Dictionnary of CPU ID to list of frequencies.
 |      |- capacity-classes (lisa.conf.IntListList): 
 |         (derived from platform-info/cpu-capacities) Capacity classes
 |         modeled by a list of CPU IDs for each capacity, sorted by
 |         capacity.
```

Reply:
Right, makes sense. Thanks for the explanation.